lib/compiler-cranelift/src/translator/code_translator.rs

```diff
@@ -1317,25 +1401,29 @@ pub fn translate_operator<FE: FuncEnvironment + ?Sized>(
         Operator::TableGet { table: index } => {
             let table_index = TableIndex::from_u32(*index);
             let table = state.get_or_create_table(builder.func, *index, environ)?;
-            let index = state.pop1();
+            let (index, _) = state.pop1();
             state.push1(environ.translate_table_get(builder, table_index, table, index)?);
         }
         Operator::TableSet { table: index } => {
             let table_index = TableIndex::from_u32(*index);
             let table = state.get_or_create_table(builder.func, *index, environ)?;
-            let value = state.pop1();
-            let index = state.pop1();
+            let (value, value_metadata) = state.pop1();
+            let (index, _) = state.pop1();
             environ.translate_table_set(builder, table_index, table, value, index)?;
+            if value_metadata.ref_counted {
+                // TODO: look into using "move"-like semantics here to avoid doing this
+                environ.translate_externref_dec(builder.cursor(), value)?;
```

Comment:
Why? Does table_set do an inc-ref? If it didn't, we wouldn't need the dec-ref here.

(Oh I guess that's what the comment is about. OK.)

Reply:
Yep, fixed locally! I agree